retrofit2/src/test/java/com/linecorp/armeria/client/http/retrofit2/ArmeriaCallFactoryTest.java

```diff
@@ -347,6 +372,91 @@ public void respectsHttpClientUri_endpointGroup() throws Exception {
                                      .build());
     }
 
+    @Test
+    public void urlAnnotation() throws Exception {
+        EndpointGroupRegistry.register("bar",
+                                       new StaticEndpointGroup(Endpoint.of("127.0.0.1", server.httpPort())),
+                                       ROUND_ROBIN);
+        Service service = new ArmeriaRetrofitBuilder()
+                .baseUrl("none+http://group:foo/")
+                .addConverterFactory(JacksonConverterFactory.create(OBJECT_MAPPER))
+                .addCallAdapterFactory(Java8CallAdapterFactory.create())
+                .build()
+                .create(Service.class);
+        Pojo pojo = service.fullUrl("http://group_bar/pojo").get();
+        assertThat(pojo).isEqualTo(new Pojo("Cony", 26));
+    }
+
+    @Test
+    public void urlAnnotation_uriWithoutScheme() throws Exception {
+        EndpointGroupRegistry.register("bar",
+                                       new StaticEndpointGroup(Endpoint.of("127.0.0.1", server.httpPort())),
+                                       ROUND_ROBIN);
+        assertThat(service.fullUrl("//localhost:" + server.httpPort() + "/nest/pojo").get()).isEqualTo(
+                new Pojo("Leonard", 21));
+        assertThat(service.fullUrl("//group_bar/nest/pojo").get()).isEqualTo(new Pojo("Leonard", 21));
+
+        assertThat(service.fullUrl("//localhost:" + server.httpPort() + "/pojo").get()).isEqualTo(
+                new Pojo("Cony", 26));
+        assertThat(service.fullUrl("//group_bar/pojo").get()).isEqualTo(new Pojo("Cony", 26));
+    }
+
+    @Test
+    public void sessionProtocolH1C() throws Exception {
+        Service service = new ArmeriaRetrofitBuilder()
+                .baseUrl("none+h1c://127.0.0.1:" + server.httpPort())
```

Comment:
Could we add `none+` automatically internally so that a user doesn't have to do that when calling `baseUrl()`?